index.js

```diff
@@ -27,73 +24,31 @@ export async function setupPlugin({ config, global, cache }) {
   );
 
   if (!statusOk(authenticationResponse)) {
-    throw new Error(`Unable to access zendesk api's for ${config.hostName}`);
+    throw new Error(`Unable to access ZenDesk API's for ${config.hostName}`);
   } else {
     console.info(
-      `Inetail setup with zendesk for ${config.hostName} successful`
+      `Initial setup with ZenDesk for ${config.hostName} successful`
     );
   }
-}
 
-function isEmail(email) {
-  const re =
-    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-  return re.test(String(email).toLowerCase());
-}
-
-function statusOk(res) {
-  return String(res.status)[0] === "2";
-}
+  global.triggeringEvents = (config.triggeringEvents || "")
+    .split(",")
+    .map((v) => v.trim());
 
-async function fetchWithRetry(
-  url,
-  options = {},
-  method = "GET",
-  isRetry = false
-) {
-  try {
-    const res = await fetch(url, { method: method, ...options });
-    return res;
-  } catch {
-    if (isRetry) {
-      throw new Error(`${method} request to ${url} failed.`);
-    }
-    const res = await fetchWithRetry(
-      url,
-      options,
-      (method = method),
-      (isRetry = true)
-    );
-    return res;
-  }
+  global.emailDomainsToIgnore = (config.ignoredEmails || "")
+    .split(",")
+    .map((v) => v.trim());
 }
 
-async function pushUserDataToZendesk(
-  email,
-  eventName,
-  sent_at,
-  global,
-  storage
-) {
-  const userId = await storage.get(email);
-
-  if (userId) {
-    const updatedHeaders = global.defaultHeaders;
-    updatedHeaders.body = JSON.stringify({
-      user: {
-        user_fields: { [eventName]: `${sent_at}` },
-      },
-    });
-
-    const retryResponse = await fetchWithRetry(
-      `${global.fetchUserUrl}/${userId}`,
-      updatedHeaders,
+export const jobs = {
+  pushUserDataToZendesk: async (request, meta) => {
```

Comment:
Let's have this job do even more. The basic event validation should be done on `onEvent` but from the storage call onwards let's do things here.